stdlib/public/runtime/static_stub.c

```diff
@@ -0,0 +1,65 @@
+#if defined(__LINUX__) && defined(__ELF__)
+#error "This only works on Linux/ELF"  // Needs testing on other ELF platforms
+#else
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+
+// This forces resolving of these weak symbols but keeps them hidden
+// externally
+const void *unused1 __attribute__ ((unused, visibility("internal"))) = pthread_self;
+const void *unused2 __attribute__ ((unused, visibility("internal"))) = pthread_key_create;
+const void *unused3 __attribute__ ((unused, visibility("internal"))) = pthread_once;
```

Comment:
In libswiftCore.a those symbols are marked as unresolved weak:
```
nm -u ~/swift-test/usr/lib/swift_static/linux/libswiftCore.a |grep ' w '|sort -u
                 w __pthread_key_create
                 w pthread_once
                 w pthread_self
                 w sched_yield
```
And this causes gold to not link them at all so they end up having a linear address of 0x0 in the binary. Im not sure if this is a gold bug or working as intended but I couldnt find a linker option to force these symbols to be resolved so I added these three lines to make them into normal unresolved symbols

```
 nm -u ~/swift-test/usr/lib/swift_static/linux/x86_64/static_stub.o 
                 U abort
                 U fprintf
                 U fputc
                 U pthread_key_create
                 U pthread_once
                 U pthread_self
                 U stderr
```